features/step_definitions/basic_steps.rb

```diff
@@ -9,3 +9,16 @@
 When(/^I click "([^"]*)"$/) do |button|
   click_link_or_button button
 end
+
+Given(/^the following restaurant exists$/) do
+   @restaurant = FactoryGirl.create(:restaurant)
+end
+
+Given(/^I am on the restaurant menu page$/) do
+  restaurant = @restaurant.id
```

Comment:
I see no need to declare a new variable here. I would use `@restaurant` in the line below.